Detectors/TPC/workflow/src/CalibdEdxSpec.cxx

```diff
@@ -38,86 +35,72 @@ class CalibdEdxDevice : public Task
  public:
   void init(framework::InitContext& ic) final
   {
-    const int slotLength = ic.options().get<int>("tf-per-slot");
-    const int maxDelay = ic.options().get<int>("max-delay");
-    const int minEntries = std::max(50, ic.options().get<int>("min-entries"));
-    const int nbins = std::max(10, ic.options().get<int>("nbins"));
-    const bool applyCuts = ic.options().get<bool>("apply-cuts");
-    const float minP = ic.options().get<float>("min-momentum");
-    const float maxP = ic.options().get<float>("max-momentum");
-    const int minClusters = std::max(10, ic.options().get<int>("min-clusters"));
-    const bool dumpData = ic.options().get<bool>("direct-file-dump");
-
-    assert(minP < maxP);
-
-    mCalibrator = std::make_unique<tpc::CalibdEdx>(nbins, minEntries, minP, maxP, minClusters);
-    mCalibrator->setApplyCuts(applyCuts);
-
-    mCalibrator->setSlotLength(slotLength);
-    mCalibrator->setMaxSlotsDelay(maxDelay);
-
-    if (dumpData) {
-      mCalibrator->enableDebugOutput("calib_dEdx.root");
+    const int minEntriesSector = ic.options().get<int>("min-entries-sector");
+    const int minEntries1D = ic.options().get<int>("min-entries-1d");
+    const int minEntries2D = ic.options().get<int>("min-entries-2d");
+
+    const int dEdxBins = ic.options().get<int>("dedxbins");
+    const int zBins = ic.options().get<int>("zbins");
+    const int angularBins = ic.options().get<int>("angularbins");
+    const float mindEdx = ic.options().get<float>("min-dedx");
+    const float maxdEdx = ic.options().get<float>("max-dedx");
+
+    const bool dumpData = ic.options().get<bool>("file-dump");
+    float field = ic.options().get<float>("field");
+
+    const auto inputGRP = o2::base::NameConf::getGRPFileName();
+    const auto grp = o2::parameters::GRPObject::loadFrom(inputGRP);
+    if (grp != nullptr) {
+      field = 5.00668f * grp->getL3Current() / 30000.;
+      LOGP(info, "Using GRP file to set the magnetic field to {} kG", field);
     }
```

Comment:
Like this, the grp would overwrite the value from the command line. Since the grp will always be required to not crash, the `field` command line parameter is useless. I guess the logic should be the other way around.
This should then be combined with some dummy value in the default parameter, like `-100`. And the checked
```c++
if (ic.options().get<float>("field") > -10) {
field = ic.options().get<float>("field");
}
```
or so.

Reply:
So, if a value is passed to the command line we don't try to read the grp file, right? Its very nice. I thought that we would get a nullptr if no grp file was found. I will change this.

Reply:
Like this
```C++
    float field = ic.options().get<float>("field");

    if (field <= -100) {
      const auto inputGRP = o2::base::NameConf::getGRPFileName();
      const auto grp = o2::parameters::GRPObject::loadFrom(inputGRP);
      if (grp != nullptr) {
        field = 5.00668f * grp->getL3Current() / 30000.;
        LOGP(info, "Using GRP file to set the magnetic field to {} kG", field);
      }
    }
```